diff --git a/cmake/external/onnxruntime_external_deps.cmake b/cmake/external/onnxruntime_external_deps.cmake
index 9effd1a2db..0877604fbc 100644
--- a/cmake/external/onnxruntime_external_deps.cmake
+++ b/cmake/external/onnxruntime_external_deps.cmake
@@ -273,7 +273,9 @@ endif()
 if (NOT WIN32)
   #nsync tests failed on Mac Build
   set(NSYNC_ENABLE_TESTS OFF CACHE BOOL "" FORCE)
-  onnxruntime_fetchcontent_makeavailable(google_nsync)
+  # onnxruntime_fetchcontent_makeavailable(google_nsync)
+  find_package(nsync_cpp REQUIRED)
+  add_library(nsync::nsync_cpp ALIAS nsync_cpp)
   if (google_nsync_SOURCE_DIR)
     add_library(nsync::nsync_cpp ALIAS nsync_cpp)
     target_include_directories(nsync_cpp PUBLIC ${google_nsync_SOURCE_DIR}/public)
@@ -303,7 +305,13 @@ FetchContent_Declare(
 )
 
 # The next line will generate an error message "fatal: not a git repository", but it is ok. It is from flatbuffers
-onnxruntime_fetchcontent_makeavailable(Protobuf nlohmann_json mp11 re2 safeint GSL flatbuffers)
+onnxruntime_fetchcontent_makeavailable(Protobuf nlohmann_json mp11 safeint GSL flatbuffers)
+
+find_package(re2)
+find_package(PkgConfig)
+pkg_check_modules(RE2 IMPORTED_TARGET re2)
+add_library(re2::re2 ALIAS PkgConfig::RE2)
+
 if(NOT flatbuffers_FOUND)
   if(NOT TARGET flatbuffers::flatbuffers)
     add_library(flatbuffers::flatbuffers ALIAS flatbuffers)
